# HG changeset patch
# User Lachlan Andrew <lachlanbis@gmail.com>
# Date 1463666049 -36000
#      Thu May 19 23:54:09 2016 +1000
# Node ID 71d8a9a0642c9dc22876ea66d14a282c88d72fc7
# Parent  767c81fe30bac5a874a24c236b13a86d3a830c2a
Fix scope of umask to avoid recursive call. (bug #47961)

* file-ops.cc (umask):
  Call ::umask (since there is no gnulib::umask), not umask.

diff --git a/liboctave/system/file-ops.cc b/liboctave/system/file-ops.cc
--- a/liboctave/system/file-ops.cc
+++ b/liboctave/system/file-ops.cc
@@ -636,17 +636,17 @@ namespace octave
 
       return status;
     }
 
     int
     umask (mode_t mode)
     {
 #if defined (HAVE_UMASK)
-      return umask (mode);
+      return ::umask (mode);
 #else
       return 0;
 #endif
     }
 
     int
     unlink (const std::string& name)
     {
